src/interface_can.cpp

```diff
@@ -45,111 +56,82 @@ CANMsgQueue can_rx_queue;
 // Protocol details:
 // https://github.com/LibreSolar/ThingSet/blob/master/can.md
 
-void can_pub_msg(data_object_t data_obj)
+static uint8_t can_node_id = CAN_NODE_ID;
+
+extern ThingSet ts;
+ThingSetCAN ts_can;
+
+
+void ThingSetCAN::init_hw()
```

Comment:
Shouldn't the initialization go to the constructor, now that we have a C++ class and no C style module?

Reply:
This is a dependency thing (we have to be sure that the passed objects constructors have been run, that requires some more thought, otherwise we may call the CAN objects method for instance before it has been constructed) and a bit of legacy (the init_hw existed already). 

I intend to add a constructor to this but even then  we would want some kind of enable/disable anyways I guess.